src/main/java/com/cloudbees/plugins/credentials/CredentialsSelectHelper.java

```diff
@@ -577,8 +578,8 @@ public WrappedCredentialsStore getContext(String token) {
          * @param token    the context's {@link ContextResolver#getToken(ModelObject)}.
          * @param store    the {@link CredentialsStore}
          */
-        public WrappedCredentialsStore(ContextResolver resolver, CredentialsProvider provider,
-                                       String token, CredentialsStore store) {
+        public WrappedCredentialsStore(@NonNull ContextResolver resolver, @NonNull CredentialsProvider provider,
+                                       @NonNull String token, CredentialsStore store) {
```

Comment:
⁉️ From the usage inside `WrappedContextResolverCredentialsProvider#getContext` I am OK with resolver and provider being nonNull, but token is not obvious (but seems to be) and store is also nonNull. Was this proposed by IntelliJ or a manual analysis? (not proposed by my IntelliJ configuration ^^)

So, why `store` is not marked in this case?

Reply:
The change reflects the annotations of the fields to the constructor, as it's more visible to the caller. While `token` (line 567) is marked with `@NonNull`, `store` (line 571) isn't. From it's usage it assumes non-null, but I wasn't confident enough to actually mark it.

Reply:
I will adjust the field at the same time, I thought you were changing the constructor by looking at the only location it was called ;)